### PR TITLE
Fix ESLint errors in the Guid, Immutable, OptionsHelper, Poller, and PropTypes helpers

### PR DESCRIPTION
**Pull request by ianoxley**
Fix whitespace issues, use `const` over `let`, don't reassign function parameters, remove useless `else` blocks, no plus-plus operator, add missing semicolons, et al.

GUID Helper - Used by input decorator, form and table row
Immutable Helper - Used across demo site store - simple regression needed
Options Helper - Used across demo site for control dropdown. e.g. color options for flash
Poller helper - Not used in application. May require Dev QA to set up.
PropTypes helper is used by Decimal but is a dev checking thing so requires Dev

### CODE REVIEW

File: src/utils/helpers/poller/poller.js [hunk 1]
```diff
@@ -41,45 +41,49 @@ import './../../promises';
  */
 export default (queryOptions, functions, options) => {
   if (!setupValid(queryOptions, functions)) {
-    return;
+    return undefined;
```
Comment (abramin) [inline src/utils/helpers/poller/poller.js]: return `null`?

Comment (ianoxley) [inline src/utils/helpers/poller/poller.js]: `return;` returns undefined, so by fixing the linting error and explicitly returning `undefined` via `return undefined;` there should be no unintended side effects.

Comment (abramin) [inline src/utils/helpers/poller/poller.js]: right, I was wondering if we wanted to return `undefined`. I suppose in this context, it's the best option

File: src/utils/helpers/poller/poller.js [hunk 2]
```diff
@@ -41,45 +41,49 @@ import './../../promises';
...
-            setTimeout(poll, options.interval);
+            funcs.conditionNotMetCallback(response);
+            pollCount += 1;
+            setTimeout(poll, opts.interval);
           }
+
+          return result;
         });
-    })();
+    }());
```
Comment (abramin) [inline src/utils/helpers/poller/poller.js]: Is this right? @ianoxley

Comment (ianoxley) [inline src/utils/helpers/poller/poller.js]: It failed http://eslint.org/docs/rules/wrap-iife without that change.

Comment (abramin) [inline src/utils/helpers/poller/poller.js]: 👍

Comment (toni-leigh) [inline src/utils/helpers/poller/poller.js]: I must admit I don't fully understand how that hasn't changed functionality

**Pull request merged**